content/blog/2021-07-06-hyperparam-tuning.md

```diff
@@ -0,0 +1,241 @@
+[get the code we're working with in this repo](https://github.com/iterative/example-get-started).
+It already has DVC set up, but you can check out
+[the Get Started docs](https://dvc.org/doc/start) if you want to know how the
+DVC pipeline was created.
+
+### Run an experiment
+
+First make sure you're in a virtual environment with a command similar to this.
+
+`python -m venv .venv`
+
+After you've cloned the repo, install all of the dependencies with this command.
+
+`pip install -r requirements.txt`
+
+You should be able to open your terminal and run an experiment with the
+following command.
+
+`dvc exp run`
+
+This will trigger the training process to run and it will record the precision
+of your model. You can check out the results of your experiment with the
+following command.
+
+`dvc exp show --no-timestamp --include-params train.n_est,train.min_split`
+
+_We're adding a few options here to make the table view clearer. We aren't
+showing timestamps and we're only looking at two hyperparameter values. You can
+run `dvc exp show` without the options to see the entire table._
+
+This will produce a table similar to this.
+
+```
+┏━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━┓
+┃ Experiment              ┃ avg_prec ┃ roc_auc ┃ train.n_est ┃ train.min_split ┃
+┡━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━┩
+│ workspace               │ 0.51682  │ 0.93819 │ 175         │ 64              │
+│ master                  │ 0.56447  │ 0.94713 │ 100         │ 64              │
+│ └── a1e8716 [exp-09074] │ 0.57333  │ 0.94801 │ 100         │ 32              │
+└─────────────────────────┴──────────┴─────────┴─────────────┴─────────────────┘
+```
+
+### Start tuning with grid search
+
+Now that you've seen how to run an experiment, we're going to write a small
+script to automate grid search for us using DVC. We'll do this by creating
+queues.
+
+A queue is how DVC allows us to create experiments that won't be run until
+later. That way we can cycle through multiple hyperparameters quickly instead of
+manually updating a config file with new hyperparameter values for each
+experiment run. The command syntax for creating queues looks like this:
+
+`dvc exp run --queue -set-param train.min_split=8`
+
+In the example queue above, we're updating the `train.min_split` value that's
+inside of the `params.yaml` file. This file holds all of the hyperparameter
+values and is where DVC looks to determine if any values have changed. With the
+command above, we're automatically updating that value in the `params.yaml`
+using a queued experiment.
+
+Now we can make the script. You can add a new file to the `src` directory called
+`tuning.py`. Inside of the file, add the following code.
+
+```python
+import subprocess
+import random
+
+# Automated grid search experiments
+n_est_values = [250, 300, 350, 400, 450, 500]
+min_split_values = [100, 110, 120, 130, 140, 150]
```

Comment:
I would recommend logarithmic/exponential growth for `min_split_values`. Something like `[8, 16, 32, 64, 128, 256]` would make more sense to me. It's not likely that you would be tuning around such a tight set of values as 100-150.